File: src/main/java/com/oshi/libgenericsettings/data/TitleSeekBarMinMaxData.java
package com.oshi.libgenericsettings.data;

import android.support.annotation.ColorRes;

/**
 * Created by udioshi on 9/11/2017.
 */
public class TitleSeekBarMinMaxData extends TitleData {

    protected int maxValue;
    protected int minValue;
    protected int progress = 0;
    protected @ColorRes int seekBarColor;
    protected @ColorRes int seekBarThumbColor;
    protected @ColorRes int minMaxColor;

    public TitleSeekBarMinMaxData(String title, int minValue, int maxValue, int progress) {
        super(title);
        this.maxValue = maxValue;
        this.minValue = minValue;

        if (progress < minValue || progress > maxValue) {
            throw new IllegalArgumentException("Progress must be greater than minValue and lower than maxValue");
        }

        this.progress = progress;
        internalCalculateProgress(progress);
    }

    public int getMaxValue() {
        return maxValue;
    }

    public int getMinValue() {
        return minValue;
    }

    public int getInternalProgress() {
        return progress;
    }

    public int getSeekBarColor() {
        return seekBarColor;
    }

    public void setSeekBarColor(int seekBarColor) {
        this.seekBarColor = seekBarColor;
    }

    public int getSeekBarThumbColor() {
        return seekBarThumbColor;
    }

    public void setSeekBarThumbColor(int seekBarThumbColor) {
        this.seekBarThumbColor = seekBarThumbColor;
    }

    public int getCalculatedMaxValue() {
        return maxValue-minValue;
    }

    public int getMinMaxColor() {
        return minMaxColor;
    }

    public void setMinMaxColor(@ColorRes int minMaxColor) {
        this.minMaxColor = minMaxColor;
    }

    public int getRealProgress() {
        return progress + minValue;
    }

    private void internalCalculateProgress(int progress) {
        this.progress = progress;
        this.progress = getCalculatedMaxValue() - (progress-minValue);
    }

    public void setInternalProgress(int progress) {
        this.progress = progress;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE_SETTINGS_TITLE_SEEKBAR_MIN_MAX;
    }
}

File: src/main/java/com/oshi/libgenericsettings/data/TitleSeekBarMinMaxData.kt
package com.oshi.libgenericsettings.data

import android.support.annotation.ColorRes

/**
 * Title, SeekBar with minimum & maximum values. Being use in layout/view_type_title_seekbar_min_max.xml
 */
class TitleSeekBarMinMaxData(title : String, var minValue : Int, var maxValue : Int, var progress : Int) : TitleData(title) {

    @ColorRes var seekBarColor = 0
    @ColorRes var seekBarThumbColor = 0
    @ColorRes var minMaxColor = 0

    init {
        if (progress < minValue || progress > maxValue) {
            throw IllegalArgumentException("Progress must be greater than minValue and lower than maxValue")
        }
        internalCalculateProgress(progress)
    }

    fun setInternalProgress(progress: Int) {
        this.progress = progress
    }

    fun getInternalProgress() : Int {
        return progress
    }

    fun internalCalculateProgress(progress: Int)  {
        this.progress = progress
        this.progress = getCalculatedMaxValue() - (progress - minValue)
    }

    fun getCalculatedMaxValue() : Int {
        return maxValue - minValue
    }

    fun getRealProgress() : Int {
        return progress + minValue
    }

    override fun getViewType(): Int {
        return VIEW_TYPE_SETTINGS_TITLE_SEEKBAR_MIN_MAX
    }

}